src/NzbDrone.Core/Parser/QualityParser.cs

```diff
@@ -137,18 +137,27 @@ public static QualityModel ParseQuality(string name, string desc, int fileBitrat
             }
 
             //Based on extension
-            if (result.Quality == Quality.Unknown && !name.ContainsInvalidPathChars())
+
+            if (!name.ContainsInvalidPathChars())
             {
-                try
+                if (result.Quality == Quality.Unknown)
                 {
-                    result.Quality = MediaFileExtensions.GetQualityForExtension(Path.GetExtension(name));
-                    result.QualitySource = QualitySource.Extension;
+                    try
+                    {
+                        result.Quality = MediaFileExtensions.GetQualityForExtension(Path.GetExtension(name));
+                        result.QualitySource = QualitySource.Extension;
+                    }
+                    catch (ArgumentException)
+                    {
+                        //Swallow exception for cases where string contains illegal
+                        //path characters.
+                    }
                 }
-                catch (ArgumentException)
+                else
                 {
-                    //Swallow exception for cases where string contains illegal
-                    //path characters.
+                    result.QualitySource = QualitySource.Extension;
```

Comment:
You are setting source to extension here even if quality is not unknown since you broke up the if statement. That seems to be why the other test is failing now.